File: src/infrastructure/entities/Seller.entity.ts
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn } from "typeorm";

@Entity("sellers")
export default class SellerEntity {
  @PrimaryGeneratedColumn("increment")
  id: number;

  @Column({ type: "varchar", length: 100, nullable: false })
  name: string;

  @Column({ type: "text", nullable: true })
  description: string;

  @Column({ type: "varchar", length: 255, nullable: true })
  logo_url: string;

  @Column({ type: "decimal", precision: 3, scale: 2, nullable: true })
  rating: number;

  @Column({ type: "varchar", length: 100, nullable: true })
  contact_email: string;

  @Column({ type: "varchar", length: 20, nullable: true })
  contact_phone: string;

  @CreateDateColumn({ type: "timestamp", default: () => "CURRENT_TIMESTAMP" })
  created_at: Date;
}

File: src/main/java/com/example/demo/user/entity/SellerEntity.java
package com.example.demo.user.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Getter
@Table(name = "sellers")
@NoArgsConstructor
public class SellerEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;
    @Column(name = "name", columnDefinition = "VARCHAR(100)", nullable = false)
    String name;
    @Column(name = "description", columnDefinition = "TEXT")
    String description;
    @Column(name = "logo_url", columnDefinition = "VARCHAR(255)")
    String logoUrl;
    @Column(name = "rating", columnDefinition = "DECIMAL(3,2)")
    BigDecimal rating;
    @Column(name = "contact_email", columnDefinition = "VARCHAR(100)")
    String contactEmail;
    @Column(name = "contact_phone", columnDefinition = "VARCHAR(20)")
    String contactPhone;
    @Column(name = "created_at", columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP")
    LocalDateTime createdAt;
}
